tests/integration/exec.go

```diff
@@ -0,0 +1,197 @@
+package integration
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/types/trace"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	maxCPU      = 2 // maximum number of CPUs to pin processes to
+	cpuForTests = 0 // CPU to pin test processes to
+)
+
+// setCPUs pins the current process to a specific CPU
+func setCPUs(id ...int) {
+	cpuMask := unix.CPUSet{}
+	for _, i := range id {
+		cpuMask.Set(i)
+	}
+	unix.SchedSetaffinity(0, &cpuMask)
+}
+
+// parseCmd parses a command string into a command and arguments
+func parseCmd(fullCmd string) (string, []string, error) {
+	pattern := `'[^']*'|\S+` // split on spaces, but not spaces inside single quotes
+	re := regexp.MustCompile(pattern)
+	vals := re.FindAllString(fullCmd, -1)
+
+	if len(vals) == 0 {
+		return "", nil, fmt.Errorf("no command specified")
+	}
+	cmd := vals[0]
+	cmd, err := exec.LookPath(cmd)
+	if err != nil {
+		return "", nil, err
+	}
+	if !filepath.IsAbs(cmd) {
+		cmd, err = filepath.Abs(cmd)
+		if err != nil {
+			return "", nil, err
+		}
+	}
+
+	args := vals
+	// remove single quotes from args, since they can confuse exec
+	for i, arg := range args {
+		args[i] = strings.Trim(arg, "'")
+	}
+
+	return cmd, args, nil
+}
+
+// handleFDs closes all fds except for stdin, stdout, stderr
+// It also redirects stdout to /dev/null
+func handleFDs() error {
+	// close range of fds
+	err := unix.CloseRange(3, ^uint(0), unix.CLOSE_RANGE_UNSHARE)
```

Comment:
This syscall was added in 5.9 and glibc 2.34. Not sure about clang support. Golang uses Syscall(SYS_CLOSE_RANGE). I wonder how this passed 5.4 tests and if it passes vanilla 5.4 kernels. Just curious.

Reply:
Really curious. SYS_CLOSE_RANGE is 463, so it should have returned an error.

Reply:
Integration tests don't run per kernel so I guess that's why (probably the machine running it is one of the latest kernels).

Reply:
You're right. That's the cause.